src/test/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImplTest.java

```diff
@@ -128,4 +129,64 @@ public void testHasOpenPRWithExistingKeyAndOpenPR() {
 
         assertTrue(pullRequestStore.hasOpenPullRequests(branchName, repository));
     }
+
+    @Test
+    public void testRemovePRWithNonExistingKey() {
+        testAddPRWithNewKey(); //there is an entry in pullRequestStore with different cacheKey
+        String newKey = "different-key";
+
+        ConcurrentMap<?, ?> oldPullRequests = pullRequestStore.getPullRequests();
```

Comment:
I think this (and a few other tests here) rely on returning the internal representation of the `pullrequestStore` we cannot do that. This test can just as well do ` pullRequestStore.removePullRequest(serverId, anotherBitbucketPullRequest);`
followed by an `assert` that the old PR is still present.